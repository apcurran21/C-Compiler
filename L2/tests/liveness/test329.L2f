(@gimmeACall
	1
	%jennysNumber <- rdi
	%count <- 0
	:call_label_ciao_8
	%compVal <- %count = 7
	cjump %compVal = 1 :call_label_ciao_10
	goto :call_label_ciao_9
	:call_label_ciao_9
	%number <- mem %jennysNumber 0
	%number <<= 1
	%number += 1
	rdi <- %number
	call print 1
	%count += 1
	%jennysNumber += 8
	goto :call_label_ciao_8
	:call_label_ciao_10
	return
)
