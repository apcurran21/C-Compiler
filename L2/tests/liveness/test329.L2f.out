(
(in
(r12 r13 r14 r15 rax rbp rbx rdi)
(%jennysNumber r12 r13 r14 r15 rax rbp rbx)
(%count %jennysNumber r12 r13 r14 r15 rax rbp rbx)
(%count %jennysNumber r12 r13 r14 r15 rax rbp rbx)
(%compVal %count %jennysNumber r12 r13 r14 r15 rax rbp rbx)
(%count %jennysNumber r12 r13 r14 r15 rbp rbx)
(%count %jennysNumber r12 r13 r14 r15 rbp rbx)
(%count %jennysNumber r12 r13 r14 r15 rbp rbx)
(%count %jennysNumber %number r12 r13 r14 r15 rbp rbx)
(%count %jennysNumber %number r12 r13 r14 r15 rbp rbx)
(%count %jennysNumber %number r12 r13 r14 r15 rbp rbx)
(%count %jennysNumber r12 r13 r14 r15 rbp rbx rdi)
(%count %jennysNumber r12 r13 r14 r15 rax rbp rbx)
(%count %jennysNumber r12 r13 r14 r15 rax rbp rbx)
(%count %jennysNumber r12 r13 r14 r15 rax rbp rbx)
(r12 r13 r14 r15 rax rbp rbx)
(r12 r13 r14 r15 rax rbp rbx)
)

(out
(%jennysNumber r12 r13 r14 r15 rax rbp rbx)
(%count %jennysNumber r12 r13 r14 r15 rax rbp rbx)
(%count %jennysNumber r12 r13 r14 r15 rax rbp rbx)
(%compVal %count %jennysNumber r12 r13 r14 r15 rax rbp rbx)
(%count %jennysNumber r12 r13 r14 r15 rax rbp rbx)
(%count %jennysNumber r12 r13 r14 r15 rbp rbx)
(%count %jennysNumber r12 r13 r14 r15 rbp rbx)
(%count %jennysNumber %number r12 r13 r14 r15 rbp rbx)
(%count %jennysNumber %number r12 r13 r14 r15 rbp rbx)
(%count %jennysNumber %number r12 r13 r14 r15 rbp rbx)
(%count %jennysNumber r12 r13 r14 r15 rbp rbx rdi)
(%count %jennysNumber r12 r13 r14 r15 rax rbp rbx)
(%count %jennysNumber r12 r13 r14 r15 rax rbp rbx)
(%count %jennysNumber r12 r13 r14 r15 rax rbp rbx)
(%count %jennysNumber r12 r13 r14 r15 rax rbp rbx)
(r12 r13 r14 r15 rax rbp rbx)
()
)

)
